src/main/java/it/eng/idsa/dataapp/handler/ArtifactMessageHandler.java

```diff
@@ -53,22 +65,13 @@ public Map<String, Object> handleMessage(Message message, Object payload) {
 
 			logger.debug("Handling message with requestedElement:" + arm.getRequestedArtifact());
 
-			// Check if requested artifact exist in self description
-			if (selfDescriptionService.artifactRequestedElementExist(arm,
-					selfDescriptionService.getSelfDescription(message))) {
-				if (isBigPayload(arm.getRequestedArtifact().toString())) {
-					payload = encodePayload == true ? encodePayload(BigPayload.BIG_PAYLOAD.getBytes())
-							: BigPayload.BIG_PAYLOAD;
-				} else {
-					payload = encodePayload == true ? encodePayload(createResponsePayload().getBytes())
-							: createResponsePayload();
-				}
+			if (((Boolean) threadService.getThreadLocalValue("wss") != null
+					&& (Boolean) threadService.getThreadLocalValue("wss"))) {
+				logger.debug("Handling message with requestedElement:" + arm.getRequestedArtifact() + " in WSS flow");
+				payload = handleWssFlow(message);
```

Comment:
Here you use message, and later you use casted arm
Can we be consistent and use one of those in the flow? Not to mix, even if it is the same "message"? Just for code readability.

Reply:
You're right, but in the current flow, in order to call the exception handler we have to pass a Message, not the specific type of it.
Since the plan in issue #XX is to use ThreadLocal for storing the message and then calling it in different places, one of such places is ExceptionHanlder, I would leave it as it is for now, and change it when working on that issue.